### PR TITLE
[FARM-490] Refactor logic to HomeViewModelProvider

### PR DESCRIPTION
**Pull request by albertmiro**
[FARM-490]

#### 📲 What

Refactor planned to move logic to Home View Model provider

#### 🤔 Why
		
Logic should not be in the view
		
#### 🛠 How
		
More in-depth discussion of the change or implementation.

#### 👀 See
		
Screenshots / external resources
		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [ ] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/model/bloc/home/HomeViewModelProvider.dart [hunk 1]
```diff
@@ -55,13 +66,13 @@ class HomeViewModelProvider implements ViewModelProvider<HomeViewModel> {
...
-    return HomeViewModel(status, _refresh, _profileRepository);
+    return HomeViewModel(status, _refresh, _profileRepository, _isDebugBuild);
   }
 
   void _refresh() {
     _accountRepository.getAuthorized().then((account) {
-      if(account != null ) {
-         account.profileRepository.getCurrent();
+      if (account != null) {
+        account.profileRepository.getCurrent();
```
Comment (albertmiro) [inline lib/model/bloc/home/HomeViewModelProvider.dart:69]: Code format, no changes here.

File: lib/model/bloc/home/HomeViewModelProvider.dart [hunk 2]
```diff
@@ -23,7 +28,14 @@ class HomeViewModelProvider implements ViewModelProvider<HomeViewModel> {
   HomeViewModel _snapshot;
   ProfileEntity _lastProfile;
   ProfileRepositoryInterface _profileRepository;
+  bool _isDebugBuild;
+
   HomeViewModelProvider(this._accountRepository);
+
+  void init(BuildContext context) {
```
Comment (LeeMatthewHiggins) [inline lib/model/bloc/home/HomeViewModelProvider.dart]: not sure about this, now the view model provider. needs the build context

File: lib/model/bloc/home/HomeViewModelProvider.dart [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
+import 'package:flutter/material.dart';
 
 import '../ViewModelProvider.dart';
 
 class HomeViewModel implements LoadableViewModel, RefreshableViewModel {
   final LoadingStatus loadingStatus;
   final Function refresh;
   final ProfileRepositoryInterface currentProfile;
-  HomeViewModel(this.loadingStatus, this.refresh, this.currentProfile);
+  final bool isDebugBuild;
```
Comment (LeeMatthewHiggins) [inline lib/model/bloc/home/HomeViewModelProvider.dart]: I would ask the specific question "debugMenuVisible", then you know exactly what this bool controls.

File: lib/model/bloc/home/HomeViewModelProvider.dart (hunk 2 above)
Comment (LeeMatthewHiggins) [reply lib/model/bloc/home/HomeViewModelProvider.dart]: push out the app config dependency. maybe just have a constructor for this provider that has the debug flag passed in on construction.

**Review by LeeMatthewHiggins — CHANGES_REQUESTED** [review]

File: lib/model/bloc/home/HomeViewModelProvider.dart (hunk 2 above)
Comment (albertmiro) [reply lib/model/bloc/home/HomeViewModelProvider.dart]: Sorted out

**Review by LeeMatthewHiggins — APPROVED** [review]

**Review by alexanishchenko88 — APPROVED** [review]

**Pull request merged**